hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -764,425 +737,22 @@ def nested_dtypes(
     ).filter(lambda d: max_itemsize is None or d.itemsize <= max_itemsize)
 
 
-@defines_strategy()
-def valid_tuple_axes(
-    ndim: int,
-    *,
-    min_size: int = 0,
-    max_size: Optional[int] = None,
-) -> st.SearchStrategy[Shape]:
-    """Return a strategy for generating permissible tuple-values for the
+valid_tuple_axes = _array_helpers.valid_tuple_axes
+valid_tuple_axes.__doc__ = f"""
+    Return a strategy for generating permissible tuple-values for the
```

Comment:
This will mutate the original object, which I don't think is the intention - I think you need to add a `valid_tuple_axes = deepcopy(_array_helpers.valid_tuple_axes)`.  Or if that fails our proxies decorator should work:

```python
@proxies(_array_helpers.valid_tuple_axes)
def valid_tuple_axes(*args, **kwargs):
    return _array_helpers.valid_tuple_axes(*args, **kwargs)
valid_tuple_axes.__doc__ = ...
```

Reply:
I've used deepcopy, hope it works. I was concerned about this when I tried it out with the Array API extra too—it didn't effect the Sphinx generation but yes I imagine it could effect some tooling or subtle behaviours.